bee-ledger/src/workers/snapshot/condition.rs

```diff
@@ -6,11 +6,11 @@ use bee_tangle::{storage::StorageBackend, Tangle};
 use crate::{types::LedgerIndex, workers::snapshot::config::SnapshotConfig};
 
 /// Reasons for skipping snapshotting.
-#[derive(Debug)]
-pub enum SnapshottingSkipReason {
-    /// Not enough data yet to create a snapshot.
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum SnapshottingSkipReason {
+    #[error("Snapshotting skipped for the next {reached_in} indexes.")]
     BelowThreshold { reached_in: u32 },
-    /// Snapshotting is deferred to a later milestone.
+    #[error("Snapshotting deferred for the next {next_in} indexes.")]
```

Comment:
Can you remind me the difference between skipped and deferred please ?

Reply:
It's weird. Good that you ask. I think both variants can be merged into one, but I need to do this with a fresh mind.

Reply:
Alright...things are clear to me now, and ... this also triggered a consistency refactor.